### ISSUE TITLE
Water mist with CNF ramp creates forrtl:severe error

### ISSUE THREAD

**Issue by IFAB-AS** [firemodels/fds#11571]
Hello. 
We are trying to model a multiple spray nozzle by setting the CNF ramp (cumulative number fraction, which gives the relative amount of particles which is smaller than a given diameter). The nozzle is activated after 1 s simulated time.
- If we start the FDS file with 1.000 particles/s (which is too low) and ``&RAMP ID='my_CNF',T=0,F=0.0/``, we get a ``forttl: severe (157): Program Exception - access violation`` message after 2 s.
- If we start the FDS file with 10.000 particles/s and ``&RAMP ID='my_CNF',T=0,F=0.0/``, we get a ``forttl: severe (157): Program Exception - access violation`` at the activation time of the nozzle (1s).
- If we increase the minimum diameter and start the FDS file with 10.000 particles/s and ``&RAMP ID='my_CNF',T=9.99,F=0.0/``, we get a ``forttl: severe (157): Program Exception - access violation`` at approx. 2s.
- If we increase the amount of particles to 100.000 particles/s and use ``&RAMP ID='my_CNF',T=9.99,F=0.0/`` we get a ``forttl: severe (157): Program Exception - access violation`` at the activation of the nozzle at 1s.

We tried:
- using different versions of FDS (6.7.9, 6.7.9-1634)
- using a mesh with higher resolution
- using up to 200.000 particles/s
- using a lower velocity for the droplets
- using a higher cut-off for the minimum droplet size (e.g. 40 micrometer, but then we neglect 40% of the smallest droplets and this might also crash at a later stage
- increasing the offset
- we also get sometimes the error type which is also described in the user guide: ``WARNING Delta TMP_G. Mesh: 5 Particle: 3245``. If we increase the particles/s in these cases, we get the ``forrtl:severe`` error.

A modified version of the nozzle is attached (as it is client data, we had to change some entries, but the errors are the same). Is it a problem in our file or with the code?

[Multiple_spray-nozzle-modified.txt](https://github.com/firemodels/fds/files/11026604/Multiple_spray-nozzle-modified.txt)

**Comment by mcgratta** [comment]
I ran the case with the latest test version and hit a seg fault just after the droplets are initialized. The traceback suggests it is related to the evaporation of very small droplets. 
```
 Time Step:     40, Simulation Time:      1.76 s
forrtl: severe (174): SIGSEGV, segmentation fault occurred
Image              PC                Routine            Line        Source
libpthread-2.17.s  00002AFC59A87630  Unknown               Unknown  Unknown
fds_impi_intel_li  000000000045F0EC  physical_function        4616  func.f90
fds_impi_intel_li  000000000045EDFF  physical_function        4636  func.f90
fds_impi_intel_li  00000000006E49C9  part_mp_particle_        3549  part.f90
fds_impi_intel_li  0000000000917BE7  MAIN__                    807  main.f90
```

**Comment by drjfloyd** [comment]
I get an error in read when I run in debug. FDS is trying to set the kill diameter but the minimum diameter in the CNF is 0, I think we need to throw an error for this.

With this CNF we could get droplets whose mass approaches the limit of precision for the EB type. May need to think about some maximum lower limit that we can meaningfully compute.  That being said I'll look to see if there is some expression where we need to reorder operations or something to avoid a SPACING issue.

**Comment by mcgratta** [comment]
This problem is the genesis of the use of the two default droplet distributions. To avoid those many tiny droplets.

**Comment by IFAB-AS** [comment]
I did start the ramp for the CNF with ``&RAMP ID='my_CNF',T=0,F=0.0/`` as this is also done in the example in the User Guide. If I set the DIAMETER for the Rosin-Rammler-Lognormal, the CNF neglects many of the smaller droplets. If I plot the CVF (cumulative volume fraction), the plots are similar as the smaller droplets also have small mass/volume. This means, if only the volume/mass is of interest, the smaller droplets are not that important. Even if they are removed by airstream, the most mass is in the larger droplets.

For us, this already solves the problem. But you might be able to define some cut-off for the minimum diameter or add some advice in the User Guide to inform the user that the smaller diameters should be not given in the RAMP.
